seed/redis/component.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2024 HUMAN Security.
+// Use of this source code is governed by a MIT style
+// license that can be found in the LICENSE file.
+
+package redis
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-redis/redis/v8"
+	"github.com/perimeterx/envite"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// ComponentType represents the type of the redis seed component.
+const ComponentType = "redis seed"
+
+// SeedComponent is a component for seeding redis with data.
+type SeedComponent struct {
+	lock   sync.Mutex
+	config SeedConfig
+	status atomic.Value
+	writer *envite.Writer
+}
+
+// NewSeedComponent creates a new SeedComponent instance.
+func NewSeedComponent(config SeedConfig) *SeedComponent {
+	r := &SeedComponent{config: config}
+	r.status.Store(envite.ComponentStatusStopped)
+	return r
+}
+
+func (r *SeedComponent) Type() string {
+	return ComponentType
+}
+
+func (r *SeedComponent) AttachEnvironment(_ context.Context, _ *envite.Environment, writer *envite.Writer) error {
+	r.writer = writer
+	return nil
+}
+
+func (r *SeedComponent) Prepare(context.Context) error {
+	return nil
+}
+
+func (r *SeedComponent) Start(ctx context.Context) error {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	r.status.Store(envite.ComponentStatusStarting)
+
+	err := r.seed(ctx)
+	if err != nil {
+		r.status.Store(envite.ComponentStatusFailed)
+		return err
+	}
+
+	r.status.Store(envite.ComponentStatusFinished)
+
+	return nil
+}
+
+func (r *SeedComponent) seed(ctx context.Context) error {
+	r.writer.WriteString("starting redis seed")
+	client, err := r.clientProvider()
+	if err != nil {
+		return err
+	}
+
+	if err = client.FlushAll(ctx).Err(); err != nil {
+		return err
+	}
+
+	if err = r.setEntries(ctx, err, client); err != nil {
+		return err
+	}
+
+	if err = r.HashSetEntries(ctx, err, client); err != nil {
+		return err
+	}
+	r.logInsertions()
+
+	return nil
+}
+
+func (r *SeedComponent) setEntries(ctx context.Context, err error, client *redis.Client) error {
+	for _, entry := range r.config.Entries {
+		err = client.Set(ctx, entry.Key, entry.Value, entry.TTL*time.Second).Err()
```

Comment:
`entry.TTL*time.Second` is a mistake, it's already a duration - multipying by time.Second will just multiply the duration by 1000^3 ([source code](https://github.com/golang/go/blob/master/src/time/time.go#L645))

Reply:
Fixed